src/main/java/com/github/firmwehr/gentle/GentleCompiler.java

```diff
@@ -173,6 +175,26 @@ private static void checkCommand(Path path) {
 
 	private static void compileCommand(Path path, CompilerBackendHandler handler) {
 		try {
+			// trigger load of firm builder class to init firm backend (required by some methods below)
+			try {
+				Class.forName(FirmBuilder.class.getName());
+			} catch (ClassNotFoundException e) {
+				throw new InternalCompilerException("firm builder class is missing, what happened?");
+			}
+
+			// generate matching filename for input and call backend handler
+			String programBaseName = FilenameUtils.removeExtension(path.getFileName().toString());
+			String assemblyFilename = programBaseName + ".s";
+			Path assemblyFile = path.resolveSibling(assemblyFilename);
+
+			// dump vcg files in directory with programm base name, so they are easier to manage
+			if (CompilerArguments.get().dumpGraphs()) {
+				var dumpBaseDir = path.resolveSibling("vcg/" + programBaseName).toFile();
+				FileUtils.forceMkdir(dumpBaseDir);
+
+				binding_irdump.ir_set_dump_path(dumpBaseDir.getPath());
+			}
+
```

Comment:
Seems fine, could be moved to its own method to hide away the ugliness but not necessary imo